### ISSUE TITLE
Unreadable directory causes difficult to debug failure

### ISSUE THREAD

**Issue by jasontibbitts** [rpm-software-management/rpm#776]
I recently helped track down a rather bizarre build failure, where rpmbuild exited 1 (causing mock to abort) but it wasn't really obvious why.  Turns out that executable permissions had been removed from a directory.  (Upstream had added a directory where previously there were only files, and so an overly broad `chmod 600 *` remained in the spec.)  There were complaints about this in various brp scripts but these didn't actually cause failures and ended up being buried in the rest of the output.  The last line output by rpmbuild itself was just `+ exit 0`.

The below simple spec should illustrate.  (Running in mock is not necessary to see the issue.)  The %prep section and %license bit in files aren't necessary, but serve to illustrate how difficult it can be to find the underlying cause.  Running with `-vv` doesn't expose any additional information at the end besides some calls to the compressor.  If you would like to see this with a less contrived spec, I can provide the package I originally helped debug.

```
Summary: foo
Name: foo
Version: 1
Release: 1%{?dist}
License: 0BSD

%description
Foo.

%prep
touch lic

%install
pushd %buildroot
mkdir -p etc/foo/bar
touch etc/foo/bar/{a,b,c}
chmod 600 etc/foo
popd

%check
echo foo

%files
%license lic
/etc/foo/

%changelog
* Fri Jun 28 2019 Jason L Tibbitts III <j@tib.bs> - 1-1
- Foo
```

**Comment by pavlinamv** [comment]
Fixed in commit 7a227533d1342dccc5b3717554a35dbe2baa9832

**Comment by jasontibbitts** [comment]
Thanks!
